src/server/scripts/Northrend/zone_grizzly_hills.cpp

```diff
@@ -934,6 +935,254 @@ class spell_infected_worgen_bite : public SpellScriptLoader
+{
+    SPELL_VEHICLE_WARHEAD_FUSE = 49107,
+    SPELL_ALLIANCE_KILL_CREDIT_TORPEDO = 49510,
+    SPELL_HORDE_KILL_CREDIT_TORPEDO = 49340,
+    NPC_HORDE_LUMBERBOAT = 27702,
+    NPC_ALLIANCE_LUMBERBOAT = 27688,
+    SPELL_DETONATE = 49250
+};
+
+class npc_rocket_propelled_warhead : public CreatureScript
+{
+public:
+    npc_rocket_propelled_warhead() : CreatureScript("npc_rocket_propelled_warhead") { }
+
+    struct npc_rocket_propelled_warheadAI : public VehicleAI
+    {
+        npc_rocket_propelled_warheadAI(Creature* creature) : VehicleAI(creature)
+        {
+            _finished = false;
+            _faction = ALLIANCE;
+        }
+
+        void PassengerBoarded(Unit* who, int8 /*seatId*/, bool apply) override
+        {
+            if (apply && who->ToPlayer())
+            {
+                DoCast(me, SPELL_VEHICLE_WARHEAD_FUSE);
+                _faction = who->ToPlayer()->GetTeamId();
+            }
+        }
+
+        void JustReachedHome() override
+        {
+            _finished = false;
+            me->SetVisible(true);
+            me->GetMotionMaster()->Clear(true);
+        }
+
+        void DoAction(int32 /*action*/) override
+        {
+            FinishQuest(false, _faction);
+        }
+
+        void SpellHit(Unit* caster, SpellInfo const* /*spellInfo*/) override
+        {
+            if (caster->GetEntry() == NPC_HORDE_LUMBERBOAT || caster->GetEntry() == NPC_ALLIANCE_LUMBERBOAT)
+                FinishQuest(true, _faction);
+        }
+
+        void FinishQuest(bool success, uint32 faction)
+        {
+            if (_finished)
+                return;
+
+            _finished = true;
+
+            if (success)
+                DoCast(me, faction == ALLIANCE ? SPELL_ALLIANCE_KILL_CREDIT_TORPEDO : SPELL_HORDE_KILL_CREDIT_TORPEDO);
+
+            DoCast(me, SPELL_DETONATE);
+            me->RemoveAllAuras();
+            me->SetVisible(false);
+            me->GetMotionMaster()->MoveTargetedHome();
+        }
+
+    private:
+        uint32 _faction;
+        bool _finished;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_rocket_propelled_warheadAI(creature);
+    }
+};
+
+enum WarheadSpells
+{
+    SPELL_WARHEAD_Z_CHECK = 61678,
+    SPELL_WARHEAD_SEEKING_LUMBERSHIP = 49331,
+    SPELL_WARHEAD_FUSE = 49181
+};
+// 49107 - Vehicle: Warhead Fuse
+class spell_vehicle_warhead_fuse : public SpellScriptLoader
+{
+public:
+    spell_vehicle_warhead_fuse() : SpellScriptLoader("spell_vehicle_warhead_fuse") { }
+
+    class spell_vehicle_warhead_fuse_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_vehicle_warhead_fuse_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARHEAD_Z_CHECK) || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_SEEKING_LUMBERSHIP) || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_FUSE))
+                return false;
+            return true;
```

Comment:
this can be simplified to:

```cpp
return !(
  !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_Z_CHECK)
  || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_SEEKING_LUMBERSHIP)
  || !sSpellMgr->GetSpellInfo(SPELL_WARHEAD_FUSE)
)
```

so no `if` involved